### PR TITLE
uploads, restricted manuals, solos, newest solves

### PR DESCRIPTION
**Pull request by Luminoscity**
1. adds the Restricted Manuals page that displays the manuals that are not allowed for any mission.
2. Users must be logged in to upload anything, user info is recorded for who uploaded it, info displayed to Verify roles on the mission and missionpack pages #98 
3. New view on user page: Newest, which sorts the missions solves by how recent, done instead of #103 
4. Solo solves filter on homepage #99 
5. Moved Proof box on Mission upload #17 
6. Improved messages when uploads fail #106

### CODE REVIEW

**Comment by Luminoscity** [comment]
1. image
2. image
3. image
4. image
5. image
6. image
image

File: src/routes/upload/_CompletionSection.svelte [hunk 1]
```diff
@@ -2,19 +2,21 @@
...
 	export let missionNames: string[];
 	export let solverNames: string[];
+	export let user: FrontendUser;
 	export let factoryStatus: { [name: string]: string | null };
 
 	let missionName: string = '';
 
 	let completion: Completion = new Completion();
 	completion.dateAdded = new Date();
+	completion.uploadedBy = user.id;
```
Comment (samfundev) [inline src/routes/upload/_CompletionSection.svelte:19]: We shouldn't pass the user ID down to the page, the server should be the one to add that information.

File: src/routes/upload/missionpack/+server.ts [hunk 2]
```diff
@@ -4,12 +4,26 @@ import type { RequestEvent } from '@sveltejs/kit';
 
 export async function POST({ request }: RequestEvent) {
 	const pack: MissionPack = await request.json();
```
Comment (samfundev) [inline src/routes/upload/missionpack/+server.ts:6]: If upload requires a user account, we should check for it on the server side.

File: src/routes/user/[user]/+page.svelte [hunk 3]
```diff
@@ -27,14 +27,17 @@
 	let completions: MissionCompletion[] = data.completions;
 	let tpMissions: Mission[] = data.tpMissions;
 
+	const dateOptions: Intl.DateTimeFormatOptions = { year: 'numeric', month: 'short', day: 'numeric' };
+	// let seeUnverif = $page.data.user !== null && $page.data.user.id == shownUser?.id || hasPermission($page.data.user, Permission.RenameUser);
+	let seeUnverif = $page.data.user !== null && $page.data.user.id == shownUser?.id;
```
Comment (samfundev) [inline src/routes/user/[user]/+page.svelte:32]: These don't seem to be used.

**Review by samfundev — CHANGES_REQUESTED** [review]

File: src/routes/upload/missionpack/+server.ts (hunk 2 above)
Comment (Luminoscity) [reply src/routes/upload/missionpack/+server.ts:10]: Please check that I did that right.

File: src/routes/user/[user]/+page.svelte (hunk 3 above)
Comment (Luminoscity) [reply src/routes/user/[user]/+page.svelte]: These will be used soon, removed for now, thanks.

File: src/routes/upload/_CompletionSection.svelte (hunk 1 above)
Comment (Luminoscity) [reply src/routes/upload/_CompletionSection.svelte]: Please check that I did that right.

**Pull request merged**